File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
# 모든 명함을 수납할 수 있는 가장 작은 지갑을 만들 때, 지갑의 크기를 return 
# 회전 가능 하므로 가로/세로 중 가장 큰 값을 뽑은 후, 세로에서 가장 큰 값이 아닌 두번째로 큰 값을 곱해주면 된다.

def solution(sizes):
    answer = 0 
    w = []
    h = []

    for size in sizes:
        if size[0]<size[1]:
            size[0], size[1]=size[1], size[0]
        w.append(size[0])
        h.append(size[1])
  
            
    return max(w)*max(h)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
// 명함 회전 가능
// 모든 명함 수납 가능한 가장 작은 크기의 지갑

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        // 각 명함의 긴 쪽을 width로, 짧은 쪽을 height로
        int maxWid = 0;
        int maxHei = 0;
        
        for(int[] card: sizes){
            int longer = Math.max(card[0], card[1]);
            int shorter = Math.min(card[0], card[1]);
            // 최댓값 갱신
            maxWid = Math.max(maxWid, longer);
            maxHei = Math.max(maxHei, shorter);
        }

        
        return maxWid*maxHei;
    }
}